frontend/packages/console-app/src/components/quick-starts/loader/QuickStartsLoader.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { QuickStart } from '../utils/quick-start-types';
+import { getQuickStarts } from '../utils/quick-start-utils';
+import QuickStartPermissionChecker from './QuickStartPermissionChecker';
+
+type QuickStartsLoaderProps = {
+  children: (quickStarts: QuickStart[], loaded: boolean) => React.ReactNode;
+};
+
+const QuickStartsLoader: React.FC<QuickStartsLoaderProps> = ({ children }) => {
+  const [quickStarts, setQuickStarts] = React.useState<QuickStart[]>([]);
+  const [allowedQuickStarts, setAllowedQuickStarts] = React.useState<QuickStart[]>([]);
```

Comment:
When an array is set, this means that the quick start permissions have been checked. When unset, it means we're still loading. It is possible that in the future we don't have any quick starts if all CRs are removed.
```suggestion
  const [allowedQuickStarts, setAllowedQuickStarts] = React.useState<QuickStart[]>(quickStarts.length ? null : []);
```

Reply:
This would still return loaded  as false all the time even if there are no CRs to begin with. I think my earlier implementation of having a separate state for loaded would be better to handle both use cases. Updated to have a separate loaded state.